proposals/NNNN-SDL-behavior-in-case of-Low-Voltage.md

```diff
@@ -0,0 +1,74 @@
+
+Implement logic that will allow SDL to resume after battery charge is restored or to start up correctly in the next ingnition cycle if SDL was shut down due to LOW VOLTAGE event. 
+
+## Proposed solution
+
+When battery voltage hits below 7v SDL will "freeze" all operation untill it will be switched off or resumed. During LOW_VOLTAGE state proposed the following SDL behavior:
+
+- SDL ignores all requests from mobile applications
+- SDL ignores all responses and messages from HMI except messages for "WAKE_UP" or "IGNITION_OFF"
+- SDL stops audio/video streaming
+- During LOW_VOLTAGE all transports are unavailable for SDL
+- SDL persists resumption related data stored before receiving LOW_VOLTAGE message
+- After WAKE_UP applications data will be resumed to the state before LOW_VOLTAGE event
+- If LOW_VOLTAGE was received at the moment of writing to policies database, SDL and Policies Manager must keep policies database correct and working. After "WAKE_UP" policy database reflects the last known correct state.
+- SDL and PoliciesManager must persist 'consumer data' (resumption-related + local PT) periodically and independently of the external events
+- SDL resumes its regular work after receiving "WAKE_UP"
+- SDL must be able to start up correctly in the next ignition cycle after it was powered off
+
+## Details of implementation
+
+To implement changes in SDL regarding LOW_VOLTAGE event it is proposed to add new element to <"ApplicationsCloseReason"> enum in HMI_API:
+
+```xml
+<enum name="ApplicationsCloseReason">
+  <description>Describes the reasons for exiting all of applications.</description>
+  <element name="IGNITION_OFF" />
+  <element name="MASTER_RESET" />
+  <element name="FACTORY_DEFAULTS" />
+  <element name="SUSPEND" />
+  <element name="LOW_VOLTAGE" />
+</enum>
+```
+
+On getting "OnExitAllApplications(LOW_VOLTAGE)", SDL must:
+  - stop all read write activities
+  - ignore all RPC from mobile side
+  - ignore all RPCs from HMI side, except "OnAppInterfaceUnregistered(IGNITION_OFF)" or "OnAwakeSDL"
+
+It is expected that in case LOW_VOLTAGE state will be during 10 seconds VMCU shall power off the CCPU and HMI will send "OnAppInterfaceUnregistered(IGNITION_OFF)" to SDL and SDL will finish its work.
+
+SDL must to start up correctly in the next ignition cycle after LOW_VOLTAGE event
+
+When battery voltage recovers, HMI will send "OnAwakeSDL", SDL must resume its regular behavior.
+SDL must resume HMILevel of connected applications according to HMILevel resumption requirements
+
+## Potential downsides
+
+During LOW_VOLTAGE event SDL won't be able to notify or unregister applictions due to unavailable transports. Data that was transferred during LOW_VOLTAGE will be lost sinse SDL won't be able to perform any read/write activities. Applications may unregister during LOW_VOLTAGE state on HeartBeat timeout expiration.
+
+## Impact on existing code
+
+SDL core must be able to support new element in "ApplicationsCloseReason" enum
```

Comment:
@dboltovskyi also it impacts Policies and resumption writing to database process.

Reply:
Accepted. Done.